infrastructure/aws/cdk/pgStacInfra.py

```diff
@@ -0,0 +1,294 @@
+import boto3
+import yaml
+from aws_cdk import (
+    RemovalPolicy,
+    Stack,
+    aws_certificatemanager,
+    aws_ec2,
+    aws_iam,
+    aws_rds,
+    aws_s3,
+    aws_lambda
+)
+from aws_cdk.aws_apigateway import DomainNameOptions
+from aws_cdk.aws_apigatewayv2_alpha import DomainName
+from constructs import Construct
+from eoapi_cdk import (
+    BastionHost,
+    PgStacApiLambda,
+    PgStacDatabase,
+    StacBrowser,
+    StacIngestor,
+    TiPgApiLambda,
+    TitilerPgstacApiLambda,
+)
+
+from config import AppConfig
+
+APIS_RUNTIME_DIR = 'runtime'
+
+class pgStacInfraStack(Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        vpc: aws_ec2.Vpc,
+        app_config: AppConfig,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            scope,
+            id=app_config.build_service_name("pgSTAC-infra"),
+            tags=app_config.tags,
+            **kwargs,
+        )
+
+        pgstac_db = PgStacDatabase(
+            self,
+            "pgstac-db",
+            vpc=vpc,
+            engine=aws_rds.DatabaseInstanceEngine.postgres(
+                version=aws_rds.PostgresEngineVersion.VER_14
+            ),
+            vpc_subnets=aws_ec2.SubnetSelection(
+                subnet_type=aws_ec2.SubnetType.PUBLIC
+                if app_config.public_db_subnet
+                else aws_ec2.SubnetType.PRIVATE_ISOLATED
+            ),
+            allocated_storage=app_config.db_allocated_storage,
+            instance_type=aws_ec2.InstanceType(app_config.db_instance_type),
+            api_code={'entry': f'{APIS_RUNTIME_DIR}/db', 'index': 'handler.py', 'handler': 'handler'},
+        )
+
+        stac_api_lambda = PgStacApiLambda(
+            self,
+            "pgstac-api",
+            api_env={
+                "NAME": app_config.build_service_name("STAC API"),
+                "description": f"{app_config.stage} STAC API",
+            },
+            api_code={'entry': f'{APIS_RUNTIME_DIR}/stac', 'index': 'handler.py', 'handler': 'handler'},
```

Comment:
I added this to show the idea, but it isn't going to work as is because we're relying on Dockerfiles in `infrastructure/aws/Dockerfiles` and the runtimes are at the root of the repo. 

Can we remove the Dockerfiles and have an accessible runtime folder with just a handler.py and requirements.txt @vincentsarago ? 

If in the future we need to pass Docker commands we can allow for  `BundlingOptions`, like we had discussed in https://github.com/developmentseed/eoapi-cdk/pull/66#discussion_r1315459296.

Reply:
Same for the other APIs below

Reply:
> Can we remove the Dockerfiles and have an accessible runtime folder with just a handler.py and requirements.txt @vincentsarago ?

sadly no! some of the packages will be to be, which is why we need the custom dockerfile. And we also have custom code which can't really fit in an handler.py